2453 canvas course linking

jwright opened the pull request:
This PR adds a new section to the sidebar for "Course Integrations" which allows an instructor to add integrations to the existing course.

The only available course integration so far is Canvas integration. This integration allows the instructor to link a Canvas course to a current GC course. This link can then be used when importing grades and assignments (the adjustment to the flow will be made in a subsequent PR).

The linked course is stored in a new table `linked_courses` and displayed when the course integrations is displayed.

Here is the workflow:

The instructor visits `/integrations` which just lists the Canvas integration for now:

<img width="1516" alt="screen shot 2016-09-27 at 3 30 07 pm" src="https://cloud.githubusercontent.com/assets/35017/18888755/afcb5cbc-84c7-11e6-826b-32fa1d4afd2b.png">

The instructor can click on "Link Canvas Course", oauth with Canvas, and then the available Canvas courses are displayed:

<img width="1518" alt="screen shot 2016-09-27 at 3 30 37 pm" src="https://cloud.githubusercontent.com/assets/35017/18888768/b85bda0a-84c7-11e6-8a87-356a1409edea.png">

The instructor is redirected back to `/integrations` and they see that the course has been linked:

<img width="1515" alt="screen shot 2016-09-27 at 3 30 46 pm" src="https://cloud.githubusercontent.com/assets/35017/18888773/bee04afa-84c7-11e6-91ed-8b94c66fb2aa.png">

The instructor can unlink the course as well:

<img width="1512" alt="screen shot 2016-09-27 at 3 30 59 pm" src="https://cloud.githubusercontent.com/assets/35017/18888777/c5d6b40c-84c7-11e6-9a27-a792cae8d5a7.png">

Closes #2453

jwright commented:
Tagging @mariehooper and @chcholman to review the UI pieces.

mariehooper commented:
@jwright - I like this workflow a lot! The only piece I see that is a little iffy is the table with 0Auth. I wonder if that might confuse the average non-tech-savvy person? Maybe we just say 'Canvas' in that name?

jwright commented:
@mariehooper That is the name of the course in Canvas 😄

mariehooper commented:
@jwright  - perfect. OMG, I am the non-tech-savvy person!! 😂

jwright commented:
@mariehooper False. I could see why it would be confusing. 😈

jwright commented:
@acousticrobot Thanks!

1. I am removing the paths for the following in near future PRs as this replaces some of the functionality:

```
ImportersController#courses
ImportersController#assignments
Assignments::ImportersController#courses
Grades::ImportersController#courses
```

2. I have fixed the issue with the class variable. I replaced it with a class instance variable which solves the breaking builds as well.
